src/services/motion/types.ts

```diff
@@ -83,11 +84,21 @@ export const encodeMotionConfiguration = (
       return obstacleDetector;
     })
   );
-  result.setPositionPollingFrequencyHz(obj.positionPollingFrequencyHz);
-  result.setObstaclePollingFrequencyHz(obj.obstaclePollingFrequencyHz);
-  result.setPlanDeviationM(obj.planDeviationM);
-  result.setLinearMPerSec(obj.linearMPerSec);
-  result.setAngularDegsPerSec(obj.angularDegsPerSec);
+  if (obj.positionPollingFrequencyHz) {
```

Comment:
It would be safer for these checks to do:

```ts
if ('positionPollingFrequencyHz' in obj) {
  result.setPositionPollingFrequencyHz(obj.positionPollingFrequencyHz)
}
```
The current check will evaluate to false if any of these props are zero, and therefore won't be set, which will cause unexpected behavior.

Reply:
with this, when linting, I run into:
```
src/services/motion/types.ts(88,42): error TS2345: Argument of type 'number | undefined' is not assignable to parameter of type 'number'.
  Type 'undefined' is not assignable to type 'number'.
```
I will change the conditionals to be:
```
if (obj.positionPollingFrequencyHz !== undefined) {
    result.setPositionPollingFrequencyHz(obj.positionPollingFrequencyHz);
}
```
this way even if any of the props are zero or false they are set.

Reply:
That works!